undefined reference to `log_output_flush` in sample.logger.syst.catalog

**Describe the bug**

Undefined reference to `log_output_flush` when building `sample.logger.syst.catalog` tests for some platforms (e.g. `nrf52840dongle_nrf52840`).

**To Reproduce**

```
scripts/twister -v -N -M -p nrf52840dongle_nrf52840 -T samples/subsys/logging/syst
```

**Expected behavior**

Test builds successfully.

**Impact**

Unable to build and run `sample.logger.syst.catalog` tests

**Logs and console output**

```
Loading Zephyr default modules (Zephyr base).
-- Application: /__w/zephyr/zephyr/samples/subsys/logging/syst
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.10") found components: Interpreter 
-- Cache files will be written to: /__w/zephyr/zephyr/.cache
-- Zephyr version: 3.2.0-rc2 (/__w/zephyr/zephyr)
-- Found west (found suitable version "0.13.1", minimum required is "0.7.1")
-- Board: nucleo_h745zi_q_m4
-- Found host-tools: zephyr 0.15.0 (/opt/toolchains/zephyr-sdk-0.15.0)
-- Found toolchain: zephyr 0.15.0 (/opt/toolchains/zephyr-sdk-0.15.0)
-- Found Dtc: /opt/toolchains/zephyr-sdk-0.15.0/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /__w/zephyr/zephyr/boards/arm/nucleo_h745zi_q/nucleo_h745zi_q_m4.dts
-- Generated zephyr.dts: /__w/zephyr/zephyr/twister-out/nucleo_h745zi_q_m4/samples/subsys/logging/syst/sample.logger.syst.catalog.deferred/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /__w/zephyr/zephyr/twister-out/nucleo_h745zi_q_m4/samples/subsys/logging/syst/sample.logger.syst.catalog.deferred/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /__w/zephyr/zephyr/twister-out/nucleo_h745zi_q_m4/samples/subsys/logging/syst/sample.logger.syst.catalog.deferred/zephyr/dts.cmake
/__w/zephyr/zephyr/twister-out/nucleo_h745zi_q_m4/samples/subsys/logging/syst/sample.logger.syst.catalog.deferred/zephyr/zephyr.dts:354.7-379.5: Warning (simple_bus_reg): /soc/can: missing or empty reg/ranges property

warning: LOG_BACKEND_UART (defined at subsys/logging/Kconfig.backends:6) was assigned the value 'y'
but got the value 'n'. Check these unsatisfied dependencies: UART_CONSOLE (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_LOG_BACKEND_UART and/or look up
LOG_BACKEND_UART in the menuconfig/guiconfig interface. The Application Development Primer, Setting
Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be helpful
too.


warning: The choice symbol LOG_BACKEND_UART_OUTPUT_SYST (defined at
subsys/logging/Kconfig.template.log_format_config:13) was selected (set =y), but no symbol ended up
as the choice selection. See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_LOG_BACKEND_UART_OUTPUT_SYST and/or look up
LOG_BACKEND_UART_OUTPUT_SYST in the menuconfig/guiconfig interface. The Application Development
Primer, Setting Configuration Values, and Kconfig - Tips and Best Practices sections of the manual
might be helpful too.

Parsing /__w/zephyr/zephyr/Kconfig
Loaded configuration '/__w/zephyr/zephyr/boards/arm/nucleo_h745zi_q/nucleo_h745zi_q_m4_defconfig'
Merged configuration '/__w/zephyr/zephyr/samples/subsys/logging/syst/prj.conf'
Merged configuration '/__w/zephyr/zephyr/samples/subsys/logging/syst/overlay_deferred.conf'
Merged configuration '/__w/zephyr/zephyr/twister-out/nucleo_h745zi_q_m4/samples/subsys/logging/syst/sample.logger.syst.catalog.deferred/twister/testsuite_extra.conf'
Configuration saved to '/__w/zephyr/zephyr/twister-out/nucleo_h745zi_q_m4/samples/subsys/logging/syst/sample.logger.syst.catalog.deferred/zephyr/.config'
Kconfig header saved to '/__w/zephyr/zephyr/twister-out/nucleo_h745zi_q_m4/samples/subsys/logging/syst/sample.logger.syst.catalog.deferred/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 12.1.0
-- The CXX compiler identification is GNU 12.1.0
-- The ASM compiler identification is GNU
-- Found assembler: /opt/toolchains/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    TC_RUNID


-- Build files have been written to: /__w/zephyr/zephyr/twister-out/nucleo_h745zi_q_m4/samples/subsys/logging/syst/sample.logger.syst.catalog.deferred
[1/144] Preparing syscall dependency handling

[2/144] Generating include/generated/version.h
-- Zephyr version: 3.2.0-rc2 (/__w/zephyr/zephyr), build: v3.2.0-rc2
[3/144] Generating misc/generated/syscalls_subdirs.trigger
[4/144] Generating misc/generated/syscalls.json, misc/generated/struct_tags.json
[5/144] Generating include/generated/syscall_dispatch.c, include/generated/syscall_list.h
[6/144] Generating include/generated/driver-validation.h
[7/144] Generating include/generated/kobj-types-enum.h, include/generated/otype-to-str.h, include/generated/otype-to-size.h
[8/144] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
[9/144] Generating include/generated/offsets.h
[10/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc32c_sw.c.obj
[11/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc32_sw.c.obj
[12/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc7_sw.c.obj
[13/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc8_sw.c.obj
[14/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/dec.c.obj
[15/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc16_sw.c.obj
[16/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/hex.c.obj
[17/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf.c.obj
[18/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/multi_heap.c.obj
[19/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/notify.c.obj
[20/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/timeutil.c.obj
[21/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/thread_entry.c.obj
[22/144] Building C object zephyr/CMakeFiles/zephyr.dir/misc/generated/configs.c.obj
[23/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/sem.c.obj
[24/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/assert.c.obj
[25/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/fdtable.c.obj
[26/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/printk.c.obj
[27/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_packaged.c.obj
[28/144] Building C object CMakeFiles/app.dir/src/main.c.obj
[29/144] Linking C static library app/libapp.a
[30/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/heap-validate.c.obj
[31/144] Generating linker_zephyr_pre0.cmd
[32/144] Building C object zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj
[33/144] Generating linker_zephyr_pre1.cmd
[34/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/bitarray.c.obj
[35/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/onoff.c.obj
[36/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/rb.c.obj
[37/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_complete.c.obj
[38/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/heap.c.obj
[39/144] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/cpu_idle.S.obj
[40/144] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/tracing/tracing_none.c.obj
[41/144] Building C object zephyr/arch/common/CMakeFiles/arch__common.dir/sw_isr_common.c.obj
[42/144] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_msg.c.obj
[43/144] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/nmi_on_reset.S.obj
[44/144] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_mgmt.c.obj
[45/144] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/mpsc_pbuf.c.obj
[46/144] Building C object zephyr/arch/common/CMakeFiles/isr_tables.dir/isr_tables.c.obj
[47/144] Linking C static library zephyr/arch/common/libisr_tables.a
[48/144] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/fault_s.S.obj
[49/144] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/exc_exit.S.obj
[50/144] Linking C static library zephyr/arch/common/libarch__common.a
[51/144] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap_helper.S.obj
[52/144] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/isr_wrapper.S.obj
[53/144] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/nmi.c.obj
[54/144] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_core.c.obj
[55/144] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/stm32h7/mpu_regions.c.obj
[56/144] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_output_syst.c.obj
[57/144] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/reset.S.obj
[58/144] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/vector_table.S.obj
[59/144] Building C object zephyr/lib/posix/CMakeFiles/lib__posix.dir/perror.c.obj
[60/144] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/common/stm32cube_hal.c.obj
[61/144] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/common/soc_config.c.obj
[62/144] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/st_stm32/stm32h7/soc_m4.c.obj
[63/144] Linking C static library zephyr/libzephyr.a
[64/144] Building C object zephyr/lib/posix/CMakeFiles/lib__posix.dir/nanosleep.c.obj
[65/144] Building C object zephyr/lib/libc/newlib/CMakeFiles/lib__libc__newlib.dir/libc-hooks.c.obj
[66/144] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap.c.obj
[67/144] Linking C static library zephyr/lib/libc/newlib/liblib__libc__newlib.a
[68/144] Building C object zephyr/drivers/timer/CMakeFiles/drivers__timer.dir/sys_clock_init.c.obj
[69/144] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/irq_init.c.obj
[70/144] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/thread.c.obj
[71/144] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/irq_manage.c.obj
[72/144] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/prep_c.c.obj
[73/144] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/fpu.c.obj
[74/144] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/fatal.c.obj
[75/144] Linking C static library zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a
[76/144] Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/common.c.obj
[77/144] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/scb.c.obj
[78/144] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/fault.c.obj
[79/144] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/thread_abort.c.obj
[80/144] Linking C static library zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a
[81/144] Building C object modules/mipi-sys-t/CMakeFiles/..__modules__debug__mipi-sys-t.dir/library/src/mipi_syst_compiler.c.obj
[82/144] Building C object zephyr/lib/posix/CMakeFiles/lib__posix.dir/pthread_common.c.obj
[83/144] Linking C static library zephyr/lib/posix/liblib__posix.a
[84/144] Building C object modules/mipi-sys-t/CMakeFiles/..__modules__debug__mipi-sys-t.dir/library/src/mipi_syst_crc32.c.obj
[85/144] Building C object modules/mipi-sys-t/CMakeFiles/..__modules__debug__mipi-sys-t.dir/library/src/mipi_syst_inline.c.obj
[86/144] Building C object modules/mipi-sys-t/CMakeFiles/..__modules__debug__mipi-sys-t.dir/library/src/mipi_syst_init.c.obj
[87/144] Building C object zephyr/drivers/interrupt_controller/CMakeFiles/drivers__interrupt_controller.dir/intc_exti_stm32.c.obj
[88/144] Linking C static library zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a
[89/144] Building C object modules/stm32/stm32cube/CMakeFiles/..__modules__hal__stm32__stm32cube.dir/stm32h7xx/soc/system_stm32h7xx.c.obj
[90/144] Building C object modules/mipi-sys-t/CMakeFiles/..__modules__debug__mipi-sys-t.dir/library/src/mipi_syst_writer.c.obj
[91/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/banner.c.obj
[92/144] Building C object zephyr/drivers/clock_control/CMakeFiles/drivers__clock_control.dir/clock_stm32_ll_h7.c.obj
[93/144] Linking C static library zephyr/drivers/clock_control/libdrivers__clock_control.a
[94/144] Building C object zephyr/drivers/timer/CMakeFiles/drivers__timer.dir/cortex_m_systick.c.obj
[95/144] Linking C static library zephyr/drivers/timer/libdrivers__timer.a
[96/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/errno.c.obj
[97/144] Building C object modules/mipi-sys-t/CMakeFiles/..__modules__debug__mipi-sys-t.dir/library/src/mipi_syst_api.c.obj
[98/144] Building C object modules/stm32/stm32cube/CMakeFiles/..__modules__hal__stm32__stm32cube.dir/stm32h7xx/drivers/src/stm32h7xx_hal_cortex.c.obj
[99/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/version.c.obj
[100/144] Linking C static library modules/mipi-sys-t/lib..__modules__debug__mipi-sys-t.a
[101/144] Building C object modules/stm32/stm32cube/CMakeFiles/..__modules__hal__stm32__stm32cube.dir/stm32h7xx/drivers/src/stm32h7xx_hal.c.obj
[102/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/device.c.obj
[103/144] Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/pinctrl_stm32.c.obj
[104/144] Linking C static library zephyr/drivers/pinctrl/libdrivers__pinctrl.a
[105/144] Building C object zephyr/drivers/gpio/CMakeFiles/drivers__gpio.dir/gpio_stm32.c.obj
[106/144] Building C object modules/stm32/stm32cube/CMakeFiles/..__modules__hal__stm32__stm32cube.dir/stm32h7xx/drivers/src/stm32h7xx_ll_utils.c.obj
[107/144] Linking C static library zephyr/drivers/gpio/libdrivers__gpio.a
[108/144] Building C object modules/stm32/stm32cube/CMakeFiles/..__modules__hal__stm32__stm32cube.dir/stm32h7xx/drivers/src/stm32h7xx_hal_rcc.c.obj
[109/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/system_work_q.c.obj
[110/144] Building C object modules/stm32/stm32cube/CMakeFiles/..__modules__hal__stm32__stm32cube.dir/stm32h7xx/drivers/src/stm32h7xx_ll_rcc.c.obj
[111/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/main_weak.c.obj
[112/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/init.c.obj
[113/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj
[114/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/kheap.c.obj
[115/144] Building C object modules/stm32/stm32cube/CMakeFiles/..__modules__hal__stm32__stm32cube.dir/stm32h7xx/drivers/src/stm32h7xx_hal_rcc_ex.c.obj
[116/144] Linking C static library modules/stm32/stm32cube/lib..__modules__hal__stm32__stm32cube.a
[117/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mutex.c.obj
[118/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mem_slab.c.obj
[119/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/idle.c.obj
[120/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/sem.c.obj
[121/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/queue.c.obj
[122/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/msg_q.c.obj
[123/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mempool.c.obj
[124/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mailbox.c.obj
[125/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/thread.c.obj
[126/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/stack.c.obj
[127/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/work.c.obj
[128/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/xip.c.obj
[129/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/condvar.c.obj
[130/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/timer.c.obj
[131/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/timeout.c.obj
[132/144] Building C object zephyr/kernel/CMakeFiles/kernel.dir/sched.c.obj
[133/144] Linking C static library zephyr/kernel/libkernel.a
[134/144] Linking C executable zephyr/zephyr_pre0.elf
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map 
: && ccache /opt/toolchains/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc  -gdwarf-4 zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  -fuse-ld=bfd  -Wl,-T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map=/__w/zephyr/zephyr/twister-out/nucleo_h745zi_q_m4/samples/subsys/logging/syst/sample.logger.syst.catalog.deferred/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a  zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a  zephyr/lib/libc/newlib/liblib__libc__newlib.a  zephyr/lib/posix/liblib__posix.a  zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/timer/libdrivers__timer.a  zephyr/drivers/pinctrl/libdrivers__pinctrl.a  modules/stm32/stm32cube/lib..__modules__hal__stm32__stm32cube.a  modules/mipi-sys-t/lib..__modules__debug__mipi-sys-t.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  zephyr/CMakeFiles/offsets.dir/./arch/arm/core/offsets/offsets.c.obj  -L"/opt/toolchains/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/thumb/v7e-m/nofp"  -L/__w/zephyr/zephyr/twister-out/nucleo_h745zi_q_m4/samples/subsys/logging/syst/sample.logger.syst.catalog.deferred/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  -no-pie  -mcpu=cortex-m4  -mthumb  -mabi=aapcs  -mfp16-format=ieee  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -Wl,--fatal-warnings  -lm  -Wl,-lc  -L"/opt/toolchains/zephyr-sdk-0.15.0/arm-zephyr-eabi/arm-zephyr-eabi"/lib/thumb/v7e-m/nofp  -Wl,-lgcc  -lc  -specs=nano.specs && cd /__w/zephyr/zephyr/twister-out/nucleo_h745zi_q_m4/samples/subsys/logging/syst/sample.logger.syst.catalog.deferred/zephyr && /usr/local/bin/cmake -E echo
/opt/toolchains/zephyr-sdk-0.15.0/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/libzephyr.a(log_output_syst.c.obj): in function `out_func':
/__w/zephyr/zephyr/subsys/logging/log_output_syst.c:48: undefined reference to `log_output_flush'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04
 - Toolchain: Zephyr SDK 0.15.0
 - Commit SHA: 2829de75d1ae88dc53bc6e85d8f8e59ac3cca8d3

--- a/subsys/logging/Kconfig.formatting
+++ b/subsys/logging/Kconfig.formatting
@@ -24,6 +24,7 @@ endmenu
 menuconfig LOG_MIPI_SYST_ENABLE
 	bool "MIPI SyS-T format output"
 	select MIPI_SYST_LIB
+	select LOG_OUTPUT
 	help
 	  Enable MIPI SyS-T format output for the logger system.
 

